Test latest jruby instead of ancient versions

deivid-rodriguez opened the pull request:
I'm trying to get the whole test suite passing locally. This helps and saves me from installing very old versions.

grosser commented on spec/wwtd_spec.rb:
```diff
@@ -182,16 +182,16 @@ def write_default_rakefile
+    it "runs latest released jruby by default" do
+      write ".travis.yml", "rvm: jruby"
+      write "Rakefile", "task(:default) { puts %Q{RUBY: \#{RUBY_ENGINE}-\#{JRUBY_VERSION}} }"
+      wwtd("").should include "RUBY: jruby-1.7.19"
```
maybe make this `"RUBY: jruby-"` so it does not fail when a newer version comes out

deivid-rodriguez replied:
Sounds reasonable!

grosser commented:
this removes tests for the -18mode logic ... so maybe just skip the test if you don't have it installed ?

deivid-rodriguez commented:
> this removes tests for the -18mode logic ... so maybe just skip the test if you don't have it installed ?

The thing is I haven't seen any specific logic for this in the code, just some for `rbx`. Can you point me there? I'd say this is handled by the version manager.

Like in my first PR, I'm not clear about the travis failure. I'd say I need to tell travis to install every required Ruby before running the tests, but if that's the problem, I don't get why the build was previously passing.

grosser commented:
https://github.com/grosser/wwtd/blob/master/lib/wwtd/ruby.rb#L107

but yeah it's not really needed anymore and also very simple + the other lines in there are also not tested ... so might as well kill the test for that ...

deivid-rodriguez commented:
I addressed your comment and I've also been experimenting with travis. Now at least I can reproduce the failures I get locally. The "--parallel" tests seem very unreliable: they hang, the leave stopped processed and hide any output/errors. Can you reproduce this yourself?

grosser commented on .travis.yml:
```diff
@@ -1,5 +1,14 @@
+  - rvm install 2.0.0
+  - rvm install 2.1.5
+  - rvm install 2.2.1
+  - rvm install jruby
```
maybe just do before_install and leave the install alone ?

grosser commented on spec/wwtd_spec.rb:
```diff
@@ -182,16 +182,16 @@ def write_default_rakefile
-      write "Rakefile", "task(:default) { puts %Q{RUBY: \#{RUBY_ENGINE}-\#{RUBY_VERSION}} }"
-      wwtd("").should include "RUBY: jruby"
+    it "runs with given jruby version" do
+      write ".travis.yml", "rvm: jruby-1.7.19"
```
won't this break again when `rvm install jruby` installs a newer jruby ? ... maybe we should just lave this test out

grosser commented:
the parallel tests are hard to tame ... maybe increase the sleep `if ['TRAVIS']`
